streams/test-utils/src/test/java/org/apache/kafka/streams/TopologyTestDriverTest.java

```diff
@@ -1832,8 +1916,10 @@ public void process(final Record<String, String> record) {
         }
     }
 
-    @Test
-    public void shouldRespectTaskIdling() {
+    @ParameterizedTest(name = "Eos enabled = {0}")
+    @MethodSource("parameters")
```

Comment:
There is a huge amount of duplication in this parameterized test given the lack of class based parameterization in JUnit 5. I wonder if there's a better way of converting this test. Thoughts?

Reply:
see https://github.com/apache/kafka/pull/9856/commits/3a3c215b9081974b642e8ab5d8cf474e797c7e05

Reply:
The downside of this non standard naming approach is that one may end up not running some tests and it would be pretty hard to spot. What if we used `EnumSource` and had an enum `EosEnabled` with two values: `True` and `False`?

Reply:
> What if we used EnumSource and had an enum EosEnabled with two values: True and False?

I assumed that we don't like duplication like ```@ParameterizedTest``` and ```XXXSource``` :)

Will apply ```EnumSource```.

Reply:
I couldn't find a way to avoid duplication and not make it brittle. If you can think of one, that would be even better.

Reply:
I feel your idea (```EnumSource```) is the best one before Junit 5 supports to add parameter at class level.

Reply:
Also, I was hoping that by using `EnumSource` with an appropriately named enum, you can avoid the string in `ParameterizedTest`. In that case, the only duplication is the `EnumSource` line, which is strongly typed.

Reply:
> In that case, the only duplication is the EnumSource line, which is strongly typed.

oh, I use ```ValueSource``` with boolean type instead. It is strongly typed as well.

Reply:
But `ValueSource` won't make it clear what `true`/`false` mean, right? That's the main advantage of using the `enum`, we can make it clear what it means.

Reply:
> But ValueSource won't make it clear what true/false mean, right? That's the main advantage of using the enum, we can make it clear what it means.

you are right. will copy that!